### DISCUSSION TITLE
GitHub Action

### DISCUSSION THREAD

**Discussion by sagark237** [community/community#139341] [Actions] [answered]
### Select Topic Area

Question

### Body

How can I implement a custom GitHub Action that triggers on a specific label being added to a pull request, and how can I use this action to run specific tests or scripts?

**Comment by Vineettiwari02** [comment, marked as answer]
c ]'To create a custom GitHub Action that triggers when a specific label is added to a pull request, follow these steps:

Create the GitHub Action Workflow:

In your repository, create a directory for your GitHub Actions if it doesn’t already exist:

".github/workflows/"

Create a new YAML file for your action, e.g., label-trigger.yml:

"name: Label Trigger Action

on:
  pull_request:
    types: [labeled]

jobs:
  run-tests:
    if: github.event.label.name == 'run-tests'  # Change 'run-tests' to your desired label
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: Set up Node.js
        uses: actions/setup-node@v2
        with:
          node-version: '14'  # Specify the Node.js version you need

      - name: Install dependencies
        run: npm install

      - name: Run tests
        run: npm test  # Replace with your specific test command
"

Add this to Yml file

Configure the Label Condition:

In the if condition of the job, replace 'run-tests' with the name of the label that should trigger the action. This ensures the job only runs when that label is added.

Define Your Test Commands:

Replace the npm test command with whatever script or tests you need to run. This could be any command depending on your project's setup (e.g., running a Python script, executing shell commands, etc.).

Commit and Push Your Changes:

After creating or modifying the workflow file, commit your changes and push them to your repository:

git add .github/workflows/label-trigger.yml
git commit -m "Add label-trigger GitHub Action"
git push

Test the Action:

To test the action, create a pull request and add the specified label. The action should trigger automatically, and you can monitor the workflow's progress in the "Actions" tab of your GitHub repository.

Additional Notes:
Permissions: Ensure that your repository has the necessary permissions set for actions to run. The action must have access to the repository's content.
Action Reusability: You can also create a reusable action by creating a separate action in the .github/actions directory, which you can then reference in your workflow.
Debugging: If the action doesn’t work as expected, check the logs in the "Actions" tab for any errors.
By following these steps, you can set up a custom GitHub Action that triggers on specific labels, allowing you to automate testing or other scripts effectively.

**Reply by sagark237** [reply]
Thanks for answering both my queries
